src/refract/JSONSchemaVisitor.cc

```diff
@@ -609,26 +549,20 @@ namespace refract
     {
         ObjectElement* props = pObj;
         RefractElements members;
+        ArrayElement::ValueType reqVals;
+        std::vector<MemberElement*> varProps; // TODO: Add variable properties processing
         RefractElements oneOfMembers;
         IncludeMembers(e, members);
 
-        for (OptionElement::ValueType::const_iterator it = members.begin() ; it != members.end() ; ++it) {
-            if (SelectElement* sel = TypeQueryVisitor::as<SelectElement>(*it)) {
-                for (SelectElement::ValueType::const_iterator it = sel->value.begin() ; it != sel->value.end() ; ++it) {
-                    JSONSchemaVisitor v(pDefs);
-                    VisitBy(*(*it), v);
-                    oneOfMembers.push_back(v.getOwnership());
-                }
-            }
-            else {
-                Visit(*this, *(*it));
```

Comment:
https://github.com/apiaryio/drafter/pull/453/files/7c7b0f4f4ce359b2347b2c736b0e0ccfc2af1b92#diff-bebd5f017208988084d265cfae286303R643 here it is